internal/venice-client-common/src/test/java/com/linkedin/venice/stats/metrics/MetricEntityStateGenericTest.java

```diff
@@ -0,0 +1,216 @@
+    when(mockOtelRepository.createAttributes(any(), any(), (Map) any())).thenCallRealMethod();
+    VeniceMetricsConfig mockMetricsConfig = Mockito.mock(VeniceMetricsConfig.class);
+    when(mockMetricsConfig.getOtelCustomDimensionsMap()).thenReturn(new HashMap<>());
+    when(mockOtelRepository.getMetricsConfig()).thenReturn(mockMetricsConfig);
+    mockMetricEntity = mock(MetricEntity.class);
+    doReturn(HISTOGRAM).when(mockMetricEntity).getMetricType();
+    Set<VeniceMetricsDimensions> dimensionsSet = new HashSet<>();
+    dimensionsSet.add(VENICE_REQUEST_METHOD); // passed as base
+    dimensionsSet.add(VENICE_STORE_NAME); // dynamically passed
+    dimensionsSet.add(VENICE_CLUSTER_NAME); // dynamically passed
+    doReturn(dimensionsSet).when(mockMetricEntity).getDimensionsList();
+    mockSensor = mock(Sensor.class);
+    baseDimensionsMap = new HashMap<>();
+    baseDimensionsMap.put(VENICE_REQUEST_METHOD, MULTI_GET_STREAMING.getDimensionValue());
+
+    // build a map of inputDimensions
+    testInputDimensions = new HashMap<>();
+    testInputDimensions.put(VENICE_STORE_NAME, "store1");
+    testInputDimensions.put(VENICE_CLUSTER_NAME, "cluster1");
+
+    AttributesBuilder attributesBuilder = Attributes.builder();
+    for (Map.Entry<VeniceMetricsDimensions, String> entry: testInputDimensions.entrySet()) {
+      attributesBuilder.put(mockOtelRepository.getDimensionName(entry.getKey()), entry.getValue()).build();
+    }
+    for (Map.Entry<VeniceMetricsDimensions, String> entry: baseDimensionsMap.entrySet()) {
+      attributesBuilder.put(mockOtelRepository.getDimensionName(entry.getKey()), entry.getValue()).build();
+    }
+    expectedAttributesForTestInputDimensions = attributesBuilder.build();
+  }
+
+  @Test
+  public void testRecordOtelMetricCounter() {
+    LongCounter longCounter = mock(LongCounter.class);
+    when(mockMetricEntity.getMetricType()).thenReturn(MetricType.COUNTER);
+
+    MetricEntityState metricEntityState =
+        new MetricEntityStateGeneric(mockMetricEntity, mockOtelRepository, baseDimensionsMap);
+    metricEntityState.setOtelMetric(longCounter);
+
+    Attributes attributes = Attributes.builder().put("key", "value").build();
+    metricEntityState.recordOtelMetric(10, attributes);
+
+    verify(longCounter, times(1)).add(10, attributes);
+  }
+
+  @Test
+  public void testRecordMetricsWithBothOtelAndTehuti() {
+    DoubleHistogram doubleHistogram = mock(DoubleHistogram.class);
+    when(mockMetricEntity.getMetricType()).thenReturn(HISTOGRAM);
+
+    MetricEntityStateGeneric metricEntityState =
+        new MetricEntityStateGeneric(mockMetricEntity, mockOtelRepository, baseDimensionsMap);
+    metricEntityState.setOtelMetric(doubleHistogram);
+    metricEntityState.setTehutiSensor(mockSensor);
+
+    // make a copy of the testInputDimensions as it will be modified in the test
+    Map<VeniceMetricsDimensions, String> testInputDimensions = new HashMap<>(this.testInputDimensions);
```

Comment:
Nit, let's pick a different name for the cloned one, otherwise it's a bit hard to tell which one is referred later